blind_75/minimum_consecutive_cards_to_pick_up.py

```diff
@@ -0,0 +1,76 @@
+# You are given an integer array cards where cards[i] represents the value of 
+# the ith card. A pair of cards are matching if the cards have the same value.
+
+# Return the minimum number of consecutive cards you have to pick up to have a 
+# pair of matching cards among the picked cards. If it is impossible to have 
+# matching cards, return -1.
+"""
+Pseudocode
+use left and right pointers
+if there are 2 or more values at the right pointer, 
+   this is valid window so see if it's smaller than previous window size
+   decrement in counter map 
+otherwise there are no more in window, delete
+   move the left pointer forward
+otherwise just move right pointer forward
```

Comment:
Hm, I'm actually not entirely sure how this works at a glance, but I trust it's correct.

My intuition for this problem was simply to just:
```
keep a `last_seen` index for each value in a map
iterate through cards:
  - calculate the width between current index and last index seen on the same value
  - update `min_size` if less than `min_size`
  - update the last index seen on the current value
```

Let me know if I'm missing anything! I feel like this is way simpler and easier to implement.